inc/Predictors/Predictor.h

```diff
@@ -31,9 +31,15 @@ namespace PCOE {
         }
 
     private:
-        std::vector<ProgEvent> events;
-        std::vector<DataPoint> sysTrajectories;
+		std::vector<ProgEvent> events; 
+		std::vector<DataPoint> sysTrajectories;
     };
+	
+	class EmptyPrediction : public Prediction {
```

Comment:
This could be a static method on the class instead of a subclass. Could also be a free function, which is more C++ish.